compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -100,10 +102,11 @@ impl<'tcx> ty::CoroutineArgs<TyCtxt<'tcx>> {
     #[inline]
     fn discriminants(
         self,
+        ty: Ty<'tcx>,
```

Comment:
The doc comment should explain the meaning of the new argument and how it relates to `self`.

Reply:
Since it is only used for the Unknown layout error, reconstruct it with Ty::new_coroutine maybe?